src/include/serial_inline.h

```diff
@@ -63,7 +72,14 @@ __insert_serial_func(WT_SESSION_IMPL *session, WT_INSERT_HEAD *ins_head, WT_INSE
      * implementations read the old value multiple times.
      */
     for (i = 0; i < skipdepth; i++) {
-        WT_INSERT *old_ins = *ins_stack[i];
+        /*
+         * We assume the value of old_ins would not change within this loop. However, the compiler
+         * may change the code in a way that it will read the old_ins from memory again.
+         *
+         * Place a compiler barrier here to avoid this issue.
+         */
+        old_ins = *ins_stack[i];
+        WT_BARRIER();
```

Comment:
Is a compiler barrier sufficient. Should we be concerned that the CPU (i.e., ARM) might speculatively prefetch future values of `*ins_stack[I]` resulting in out of order reads of different pointer values.

Reply:
After the line, we have an atomic cas, which is a barrier. Therefore, I don't think we need anything extra here.

Reply:
Ah. I didn't notice the CAS. Thanks!